Default dimmable brightness to 255 from 100

dethpickle opened the pull request:
**Changes default ISY dimmable light brightness to 255**

Note, the branch name was misspelled and should have been "isy-brightness-fix". Thank you, auto-correct.

**Related issue:** fixes #4619

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [X] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51

Full brightness for ISY dimmers is 255. The current 100 value turns dimmer switches on to just under half brightness.  Probably just an oversight from the Sept implementation.

dethpickle committed:
Default dimmable brightness to 255 from 100

Full brightness for ISY dimmers is 255. The current 100 value turns dimmer switches on to just under half brightness.  Probably just an oversight from the Sept implementation.

jbcodemonkey commented:
PyISY lists the brightness as optional in the 'on' command.
https://github.com/automicus/PyISY/blob/master/PyISY/Nodes/node.py line 91.

Insteon devices can have a default on level, which may not be 100%. If we always send a value, we override the default device behavior, which may be undesirable.

Rather than force a level, we should just send the instruction.

dethpickle commented:
I didn't add the brightness in there.  The HA switches worked fine for me until September.  I only recently bothered to find out why they were going to half brightness now though and found the 100 in there.  That value SHOULD be overwritten by any value passed to it.  My thought is that the default value when no value is passed should be full brightness and not 39% brightness.

dethpickle commented:
So, to finish that thought, perhaps you're suggesting that removing the brightness= altogether is a better solution.

I'd plead that I barely know enough to comment on that.  I believe Teagen added it in September.  I'd wager it was a mistake that he put 100 in, rather that 255.  But if you're saying having nothing there is the same as having brightness=255, then perhaps that's a better way to go.

I've tried adding the optional `brightness=255` to my `customization:` on lights to override the 100 and it's never worked.  Scenes with explicit brightness always work.  So, it's possible what you're saying is going the right way.  I can experiment with that if you want and adjust this, if necessary.

What do you think?

jbcodemonkey commented:
Check this out:
https://github.com/jbcodemonkey/home-assistant/blob/master/homeassistant/components/light/isy994.py

Off has been using the fastoff command, no idea why. Off allows the device's ramp rate to be used and should be preferred.
On does not need a brightness setting, removing it works as expected, just tested.

Teagan42 commented:
1. 100 vs 255 - that was a misunderstanding as my insteon switches/lights
all worked great with 100, but that may have been just the lights/modules I
was using. I agree we should be using 255 or the device's on-level setting
in the ISY994 hub (so either pass brightness=255 or no brightness at all)
2. I had some issues with the not-fast off() method - I don't remember what
the issue was, but that could have just been the device's ramp rate was not
set to instant.

And - yeah, I basically rewrote the entire ISY994 integration for HA back
in September because I was tired of it not picking up devices/device types
and locks/covers did not work except for stupid "program switches" - if you
guys have any questions, let me know. I started a new company right after
releasing the ISY994 updates and haven't really had time to address
anyone's concerns, but I tried to architect it in a way that made logical
sense (at least to me). I am currently working on HA's ISY994 climate
(thermostat) integration - waiting on a PR for PyISY library though.

On Mon, Nov 28, 2016 at 4:19 PM, jbcodemonkey <notifications@github.com>
wrote:

> Check this out:
> https://github.com/jbcodemonkey/home-assistant/blob/master/homeassistant/
> components/light/isy994.py
>
> Off has been using the fastoff command, no idea why. Off allows the
> device's ramp rate to be used and should be preferred.
> On does not need a brightness setting, removing it works as expected, just
> tested.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/4621#issuecomment-263426373>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC2fZb0pFn0It8dTPMwgvyDw4CrSTTVPks5rC2FkgaJpZM4K-Utj>
> .
>

jbcodemonkey commented on homeassistant/components/light/isy994.py:
```diff
@@ -64,7 +64,7 @@ def turn_off(self, **kwargs) -> None:
             _LOGGER.debug('Unable to turn on light.')
 
-    def turn_on(self, brightness=100, **kwargs) -> None:
+    def turn_on(self, brightness=255, **kwargs) -> None:
```
Would you mind explaining what is happening here? I don't know enough to follow the logic.

dethpickle replied:
Switching to `None` works for getting the dimmables to full brightness on the standard screen toggles as well as changing `.fastoff()` to `.off()` does respect the ramp.   So, with regard to this pull request, I'll make that change.

I do think there is still an outstanding issue which I can't put my finger on (I can make it later).  That is, even with my original issue here, I'd try overriding it by putting a `brightness=255` in a `customization:` block,  but to no affect.  That is still true.  Even with `None`, the behavior is not as would be expected.  

With `None`, if you toggle a dimmable light it goes from 0 to 255 brightness.  If you then customize the light to have a `brightness=123` and try again, you'll find that the light will actually go to full brightness but the slider (when you disclose the slider by clicking on the light name) will properly show your 123.  If you then grab the slider and move it, it dims the light to the sliders set value.  This is just like I said similarly in #4619.  The issue there must be tied to something deeper.

So, I believe it's still useful (to me at least) to amend this PR to get the brightness full when toggling the interface switches.  I can also make a new issue for the brightness customization not having an effect, if you want.  Or, would you think it's best to hold it all up until it's all fixed?

Teagan42 commented:
I'd say yes. Update the PR to at least fix the issue of 100 not meaning
100% (that was my bad when I wrote it).

As for respecting the customization: I had no idea that was even a
customization option! HA has so many little niche features, but respecting
customization should be a separate issue/PR.

On Mon, Nov 28, 2016 at 5:56 PM, DaveSergeant <notifications@github.com>
wrote:

> *@dethpickle* commented on this pull request.
> ------------------------------
>
> In homeassistant/components/light/isy994.py
> <https://github.com/home-assistant/home-assistant/pull/4621>:
>
> > @@ -64,7 +64,7 @@ def turn_off(self, **kwargs) -> None:
>          if not self._node.fastoff():
>              _LOGGER.debug('Unable to turn on light.')
>
> -    def turn_on(self, brightness=100, **kwargs) -> None:
> +    def turn_on(self, brightness=255, **kwargs) -> None:
>
> Switching to None works for getting the dimmables to full brightness on
> the standard screen toggles as well as changing .fastoff() to .off() does
> respect the ramp. So, with regard to this pull request, I'll make that
> change.
>
> I do think there is still an outstanding issue which I can't put my finger
> on (I can make it later). That is, even with my original issue here, I'd
> try overriding it by putting a brightness=255 in a customization: block,
> but to no affect. That is still true. Even with None, the behavior is not
> as would be expected.
>
> With None, if you toggle a dimmable light it goes from 0 to 255
> brightness. If you then customize the light to have a brightness=123 and
> try again, you'll find that the light will actually go to full brightness
> but the slider (when you disclose the slider by clicking on the light name)
> will properly show your 123. If you then grab the slider and move it, it
> dims the light to the sliders set value. This is just like I said similarly
> in #4619 <https://github.com/home-assistant/home-assistant/issues/4619>.
> The issue there must be tied to something deeper.
>
> So, I believe it's still useful (to me at least) to amend this PR to get
> the brightness full when toggling the interface switches. I can also make a
> new issue for the brightness customization not having an effect, if you
> want. Or, would you think it's best to hold it all up until it's all fixed?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/4621>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AC2fZQiKSqrrLxIYeS5V7md_ZwaOgqXmks5rC3g2gaJpZM4K-Utj>
> .
>

dethpickle commented:
Alright, I'll take care of that and resubmit it tomorrow. Btw, I've never done anything like this so I might have to stumble through it.  Thanks!

dethpickle committed:
Brightness change for turn_on, ramp for turn_off.

Per discussion with Teagan42 and jbcodemonkey, the brightness should rightfully be None and not an explicit value.  There is a continuing issue that the ISY modules don't respect HA's brightness customization values.  A new issue will be opened for this.
Additionally, turn_off was using ISY's fastoff() which didn't respect the ramping time.  The default behavior should just be off().

dethpickle commented:
Yay! 
I'll get that residual customization issue created later today.
